### PR TITLE
mmds: add integration test for mmds guest hang

### PR DESCRIPTION
**Pull request by georgepisaltu**
## Reason for This PR

#2017 

## Description of Changes

Added an integration test which checks whether the guest mmds can hang after receiving a malformed request with a content length header value larger than the actual size of the payload.

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] The reason for this PR is clearly provided (issue no. or explanation).
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any newly added `unsafe` code is properly documented.
- [ ] Any API changes are reflected in `firecracker/swagger.yaml`.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

### CODE REVIEW

File: tests/integration_tests/functional/test_mmds.py [hunk 1]
```diff
@@ -393,3 +393,64 @@ def test_mmds_dummy(test_microvm_with_ssh):
...
+    _, stdout, stderr = ssh_connection.execute_command(cmd)
+    _assert_out(stdout, stderr, '')
+
+    # Test for a GET request with a content length longer than
+    # the actual length of the body.
+    cmd = 'curl -m 2 -s'
+    cmd += ' -X GET'
+    cmd += ' -H  "Content-Length: 100"'
+    cmd += ' -H "Accept: application/json"'
+    cmd += ' -d "some body"'
```
Comment (iulianbarbu) [inline tests/integration_tests/functional/test_mmds.py:441]: nit: add the payload to a variable.

File: tests/integration_tests/functional/test_mmds.py [hunk 2]
```diff
@@ -393,3 +393,64 @@ def test_mmds_dummy(test_microvm_with_ssh):
...
+
+    cmd = 'ip route add 169.254.169.254 dev eth0'
+    _, stdout, stderr = ssh_connection.execute_command(cmd)
+    _assert_out(stdout, stderr, '')
+
+    # Test for a GET request with a content length longer than
+    # the actual length of the body.
+    cmd = 'curl -m 2 -s'
+    cmd += ' -X GET'
+    cmd += ' -H  "Content-Length: 100"'
```
Comment (iulianbarbu) [inline tests/integration_tests/functional/test_mmds.py:439]: nit: compute the content length based on the payload length.

**Review by iulianbarbu — APPROVED** [review]

File: tests/integration_tests/functional/test_mmds.py (hunk 2 above)
Comment (georgepisaltu) [reply tests/integration_tests/functional/test_mmds.py:439]: For the purpose of this test, `Content-Length` is supposed to be higher than the actual size of the payload.

Comment (iulianbarbu) [reply tests/integration_tests/functional/test_mmds.py:439]: Indeed. You can add a default number to the payload length, to have it bigger, but in the end, all is going to be computed based on the payload.

**Review by acatangiu — APPROVED** [review]

**Pull request merged**